src/main/java/io/leangen/graphql/generator/mapping/core/PublisherAdapter.java

```diff
@@ -103,8 +103,8 @@ public void onNext(R result) {
 
             @Override
             public void onError(Throwable error) {
-                ExceptionWhileDataFetching wrapped = new ExceptionWhileDataFetching(step.getPath(), error, step.getField().getSourceLocation());
-                promise.complete(new DataFetcherResult<>(buffer, Collections.singletonList(wrapped)));
+                DataFetchingException wrapped = new DataFetchingException(step.getPath(), error, step.getField().getSourceLocation());
+                promise.completeExceptionally(wrapped);
```

Comment:
This loses the partial result :( graphql-java has no logic to unwrap `DataFetcherResult` on an exceptional completion...

Reply:
Yes... that's because CompletableFuture supposed to return only one result always and error here means that there is nothing to return actually.
Maybe it will be better to fix graphql-java to handle results with errors correctly, for example add cause exception to AbsoluteGraphQLError.
You can close this PR

Reply:
I think this was finally fixed in graphql-java 12 actually. I have to upgrade...